build w/ system boost & portaudio fails w/ `boost::tie is not a member of boost`

see related #7118 

## Environment

* SuperCollider version:  found in 3.14.0, still failing at latest HEAD
* Operating system: Gentoo
* Other details (Qt version, audio driver, etc.): boost 1.88.0

## Steps to reproduce

```sh
cmake -DSYSTEM_BOOST=ON -DAUDIOAPI=portaudio
make supernova
```

## Expected vs. actual behavior

```
server/supernova/server/server.cpp: In function ‘bool nova::set_realtime_priority(int)’:
server/supernova/server/server.cpp:278:16: error: ‘tie’ is not a member of ‘boost’; did you mean ‘xtime’?
  278 |         boost::tie(min, max) = thread_priority_interval_rt();
      |                ^~~
      |                xtime
```

--- a/lang/LangPrimSource/PyrFilePrim.cpp
+++ b/lang/LangPrimSource/PyrFilePrim.cpp
@@ -1362,22 +1362,15 @@ int prPipeOpen(struct VMGlobals* g, int numArgsPushed) {
 
     PyrFile* pfile = reinterpret_cast<PyrFile*>(slotRawObject(callerSlot));
 
-    // c++17 structured binding declarations will make this into a single line
-    // auto [error, string] = ...
-    int error;
-    std::string commandLine;
-    std::tie(error, commandLine) = slotStrStdStrVal(commandLineSlot);
-    if (error != errNone)
-        return error;
+    auto [errorCmd, commandLine] = slotStrStdStrVal(commandLineSlot);
+    if (errorCmd != errNone)
+        return errorCmd;
 
-    std::string mode;
-    std::tie(error, mode) = slotStrStdStrVal(modeSlot);
-    if (error != errNone)
-        return error;
+    auto [errorMode, mode] = slotStrStdStrVal(modeSlot);
+    if (errorMode != errNone)
+        return errorMode;
 
-    pid_t pid;
-    FILE* file;
-    std::tie(pid, file) = sc_popen_shell(std::move(commandLine), mode);
+    auto [pid, file] = sc_popen_shell(std::move(commandLine), mode);
     if (file != nullptr) {
         SetPtr(&pfile->fileptr, file);
         SetInt(callerSlot, pid);
@@ -1413,22 +1406,15 @@ int prPipeOpenArgv(struct VMGlobals* g, int numArgsPushed) {
     if (argsColl->size < 1)
         return errFailed;
 
-    // c++17 structured binding declarations will make this into a single line
-    // auto [error, mode] = ...;
-    int error;
-    std::string mode;
-    std::tie(error, mode) = slotStrStdStrVal(modeSlot);
-    if (error != errNone)
-        return error;
+    auto [errorMode, mode] = slotStrStdStrVal(modeSlot);
+    if (errorMode != errNone)
+        return errorMode;
 
-    std::vector<std::string> strings;
-    std::tie(error, strings) = PyrCollToVectorStdString(argsColl);
-    if (error != errNone)
-        return error;
+    auto [errorStr, strings] = PyrCollToVectorStdString(argsColl);
+    if (errorStr != errNone)
+        return errorStr;
 
-    pid_t pid;
-    FILE* file;
-    std::tie(pid, file) = sc_popen_argv(strings, mode);
+    auto [pid, file] = sc_popen_argv(strings, mode);
 
     if (file != nullptr) {
         SetPtr(&pfile->fileptr, file);

--- a/lang/LangPrimSource/PyrUnixPrim.cpp
+++ b/lang/LangPrimSource/PyrUnixPrim.cpp
@@ -145,9 +145,7 @@ int prString_POpen(struct VMGlobals* g, int numArgsPushed) {
     PyrSlot* callerSlot = g->sp - 1;
     PyrSlot* postOutputSlot = g->sp;
 
-    int error;
-    std::string cmdline;
-    std::tie(error, cmdline) = slotStrStdStrVal(callerSlot);
+    auto [error, cmdline] = slotStrStdStrVal(callerSlot);
     if (error != errNone)
         return error;
 
@@ -156,9 +154,7 @@ int prString_POpen(struct VMGlobals* g, int numArgsPushed) {
     return errNone;
 #endif
 
-    pid_t pid;
-    FILE* stream;
-    std::tie(pid, stream) = sc_popen_shell(std::move(cmdline), "r");
+    auto [pid, stream] = sc_popen_shell(std::move(cmdline), "r");
     if (stream != nullptr) {
         SC_Thread thread(std::bind(string_popen_thread_func, pid, stream, IsTrue(postOutputSlot)));
         thread.detach();
@@ -189,15 +185,11 @@ int prArrayPOpen(struct VMGlobals* g, int numArgsPushed) {
     if (argsColl->size < 1)
         return errFailed;
 
-    int error;
-    std::vector<std::string> strings;
-    std::tie(error, strings) = PyrCollToVectorStdString(argsColl);
+    auto [error, strings] = PyrCollToVectorStdString(argsColl);
     if (error != errNone)
         return error;
 
-    pid_t pid;
-    FILE* stream;
-    std::tie(pid, stream) = sc_popen_argv(strings, "r");
+    auto [pid, stream] = sc_popen_argv(strings, "r");
     if (stream != nullptr) {
         SC_Thread thread(std::bind(string_popen_thread_func, pid, stream, IsTrue(postOutputSlot)));
         thread.detach();

--- a/lang/LangSource/PyrObject.cpp
+++ b/lang/LangSource/PyrObject.cpp
@@ -2780,9 +2780,7 @@ std::tuple<int, std::vector<std::string>> PyrCollToVectorStdString(PyrObject* co
     for (int i = 0; i < coll->size; ++i) {
         PyrSlot argSlot;
         getIndexedSlot(coll, &argSlot, i);
-        int error;
-        std::string string;
-        std::tie(error, string) = slotStrStdStrVal(&argSlot);
+        auto [error, string] = slotStrStdStrVal(&argSlot);
         if (error != errNone) {
             strings.clear();
             return std::make_tuple(error, strings);

--- a/server/supernova/server/server.cpp
+++ b/server/supernova/server/server.cpp
@@ -24,6 +24,8 @@
 #include "nova-tt/thread_priority.hpp"
 #include "nova-tt/name_thread.hpp"
 
+#include <tuple>
+
 #include "server.hpp"
 #include "sync_commands.hpp"
 
@@ -280,8 +282,7 @@ static bool set_realtime_priority(int thread_index) {
 #    elif defined(_WIN32)
         int priority = thread_priority_interval_rt().second;
 #    else
-        int min, max;
-        boost::tie(min, max) = thread_priority_interval_rt();
+        auto [min, max] = thread_priority_interval_rt();
         int priority = max - 3;
         priority = std::max(min, priority);
 #    endif

